added venv quickstart info (#2417)
python-poetry/poetry: docs/docs/basic-usage.md

It will automatically find a suitable version constraint **and install** the package and subdependencies.

## Using your virtual environment

> By default, poetry creates a virtual environment in `{cache-dir}/virtualenvs` (`{cache-dir}\virtualenvs` on Windows).

Reviewer: Might be a good idea to reference the default value of `cache-dir` here. And also add a note about the ability to configure `venv.in-project` and how that changes this.

Author: I linked to their headers on the config doc's page. The links aren't working from GitHub but they have the same format as the other links that are working on the poetry website.

There are several ways to run commands within this virtual environment.

### Using `poetry run`

To run your script simply use `poetry run python your_script.py`.
Likewise if you have command line tools such as `pytest` or `black` you can run them using `poetry run pytest`.

### Activating the virtual environment

> The easiest way to activate the virtual environment is `poetry shell`.

Reviewer 2: Could you please add a hint, that `poetry shell` will spawn a new shell and activates there the venv? This is confusing a lot of people. Because you have to type `exit` to leave this shell and deactivate the venv.

Reviewer: You can use the details here regarding how we need to create a sub-shell in order to activate the environment from within a python application.

https://github.com/python-poetry/poetry/issues/2792#issuecomment-672730038

> Alternatively you can manually activate the virtual environment by running `source {path_to_venv}/bin/activate` (`source {path_to_venv}\Scripts\activate.bat` on Windows).

Reviewer: Might be good to move this to a table; and also perhaps referencing `source $(poetry env info --path)/bin/activate` for convinience. And maybe mention that this is required if you do not want a new sub-shell.

Author: I added that one-liner and the table.

## Using your virtual environment

By default, poetry creates a virtual environment in `{cache-dir}/virtualenvs` (`{cache-dir}\virtualenvs` on Windows).
You can change the [`cache-dir`](/docs/configuration/#cache-dir) value by editing the poetry config.
Additionally, you can use the [`virtualenvs.in-project`](/docs/configuration/#virtualenvs.in-project) configuration variable
to create virtual environment within your project directory.

There are several ways to run commands within this virtual environment.

### Using `poetry run`

To run your script simply use `poetry run python your_script.py`.
Likewise if you have command line tools such as `pytest` or `black` you can run them using `poetry run pytest`.

### Activating the virtual environment

The easiest way to activate the virtual environment is to create a new shell with `poetry shell`.
To deactivate the virtual environment and exit this new shell type `exit`.
To deactivate the virtual environment without leaving the shell use `deactivate`.

!!!note

> 	**Why a new shell?**
>  	Python cannot set or change environment variables in the current shell,
> 	Therefore, poetry has to create a new shell with the `PYTHONHOME` and `PATH`
> 	environment variables pointing to the virtual environment.
> 

Reviewer: ```suggestion
	**Why a new shell?**
	Child processes inherit their environment from their parents, but do not share 
	them. As such, any modifications made by a child process, is not persisted after 
	the child process exits. A Python application (Poetry), being a child process, 
	cannot modify the environment of the shell that it has been called from such 
	that an activated virtual environment remains active after the Poetry command 
	has completed execution.
	
	Therefore, Poetry has to create a sub-shell with the virtual envrionment activated
	in order for the subsequent commands to run from within the virtual environment.

```

Alternatively, to avoid creating a new shell, you can manually activate the
virtual environment by running `source {path_to_venv}/bin/activate` (`source {path_to_venv}\Scripts\activate.bat` on Windows).
To get the path to your virtual environment run `poetry env info --path`.
You can also combine these into a nice one-liner, `source $(poetry env info --path)/bin/activate`
To deactivate this virtual environment simply use `deactivate`.

> |                   | Linux/Mac                                      | Windows                                     | Exit/Deactivate |

Reviewer: ```suggestion
|                   | POSIX Shell                                      | Windows                                     | Exit/Deactivate |
```

|-------------------|------------------------------------------------|---------------------------------------------|-----------------|
| New Shell         | `poetry shell`                                 | `poetry shell`                              | `exit`          |
| Manual Activation | `source {path_to_venv}/bin/activate`           | `source {path_to_venv}\Scripts\activate.bat`| `deactivate`    |

> | One-liner         | `source $(poetry env info --path)/bin/activate`|                                             | `deactivate`    |

Reviewer: ```suggestion
| One-liner         | ```source`poetry env info --path`/bin/activate```|                                             | `deactivate`    |
```
This might be more compliant with posix.
